src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_UsingStatement.cs

```diff
@@ -422,9 +423,27 @@ private BoundExpression GenerateDisposeCall(SyntaxNode syntax, BoundExpression d
             }
             else
             {
-                disposeCall = methodOpt.IsExtensionMethod
-                   ? BoundCall.Synthesized(syntax, receiverOpt: null, methodOpt, disposedExpression)
-                   : BoundCall.Synthesized(syntax, disposedExpression, methodOpt);
+                BoundExpression receiver = disposedExpression;
```

Comment:
Thanks @cston. The ambiguous extension methods revealed a nasty bug, where we weren't lowering extension methods with optional parameters correctly. 

Could you take a quick look at the changes here?